### PR TITLE
allow the SAML key and certs to be given in simple Base64 format

### PR DESCRIPTION
**Pull request by briskt**
### Changed (non-breaking)
- Added flexibility to specify SAML certificates and private key in simple Base64 format as well as PEM format.

### CODE REVIEW

File: application/auth/saml/provider_test.go [hunk 1]
```diff
@@ -199,35 +198,52 @@ func Test_getRsaPrivateKey(t *testing.T) {
...
+			wantErr:      true,
+			wantNumBytes: 0,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := pemToBase64(tt.pemStr)
+			got, err := decodeKey(tt.pemStr, tt.expectedType)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("pemToBase64() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (Baggerone) [inline application/auth/saml/provider_test.go:242]: These error messages need to be updated, right? ("pemToBase64"  ---> "decideKey")

File: application/auth/saml/provider_test.go [hunk 2]
```diff
@@ -199,35 +198,52 @@ func Test_getRsaPrivateKey(t *testing.T) {
...
+		},
+		{
+			name:         "valid PEM key",
+			pemStr:       ValidPrivateKey,
+			expectedType: "PRIVATE KEY",
+			wantErr:      false,
+			wantNumBytes: 1794,
+		},
+		{
+			name:         "wrong type",
```
Comment (Baggerone) [inline application/auth/saml/provider_test.go:231]: It seems like there should be a test case for something that is already just a base64 string. (Or am I being blind to it :-)  )

File: application/auth/saml/provider.go [hunk 3]
```diff
@@ -156,16 +155,11 @@ func getCertStore(cert string) (dsig.MemoryX509CertificateStore, error) {
...
+		return certStore, errors.New("a valid PEM or base64 encoded certificate is required")
 	}
 
-	cert, err := pemToBase64(cert)
-	if err != nil {
-		return certStore, err
-	}
-
-	certData, err := base64.StdEncoding.DecodeString(cert)
+	certData, err := decodeKey(cert, "CERTIFICATE")
```
Comment (Baggerone) [inline application/auth/saml/provider.go:162]: If you aren't in too much of a hurry, you could make these type strings each a `const`.

**Review by Baggerone — APPROVED** [review]
Approved assuming you consider my remarks.

File: application/auth/saml/provider_test.go (hunk 2 above)
Comment (briskt) [reply application/auth/saml/provider_test.go:233]: Nope. I remember having a nagging feeling that I forgot something. I get that a lot.

**Pull request merged**